rule/redundant-import-alias.go

```diff
@@ -38,15 +48,69 @@ func (*RedundantImportAlias) Name() string {
 	return "redundant-import-alias"
 }
 
-func getImportPackageName(imp *ast.ImportSpec) string {
-	const pathSep = "/"
-	const strDelim = `"`
+func (r *RedundantImportAlias) checkRedundantAliases(node ast.Node) map[string]string {
+
+	var aliasedPackages = make(map[string]string)
+
+	// First pass: Identify all aliases and their usage - by default alias is redundant
+	ast.Inspect(node, func(n ast.Node) bool {
+		imp, ok := n.(*ast.ImportSpec)
+		if !ok {
+			return true
+		}
+
+		if imp.Name != nil && imp.Path != nil {
+			aliasedPackages[imp.Name.Name] = "redundant"
+		}
+		return true
+	})
+
+	if r.ignoreUsed {
```

Comment:
I'd invert the condition to reduce the indentation.

Reply:
done